src/Aspire.Hosting.Azure/Extensions/AzureOpenAIExtensions.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Nodes;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Azure;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding the Azure OpenAI resources to the application model.
+/// </summary>
+public static class AzureOpenAIExtensions
+{
+    /// <summary>
+    /// Adds an Azure OpenAI resource to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{AzureOpenAIResource}"/>.</returns>
+    public static IResourceBuilder<AzureOpenAIResource> AddAzureOpenAI(this IDistributedApplicationBuilder builder, string name)
+    {
+        var resource = new AzureOpenAIResource(name);
+        return builder.AddResource(resource)
+                      .WithParameter("name", resource.CreateBicepResourceName())
+                      .WithParameter("deployments", () => new JsonArray(resource.Deployments.Select(x => x.ToJsonNode()).ToArray()))
+                      .WithParameter(AzureBicepResource.KnownParameters.PrincipalId)
+                      .WithParameter(AzureBicepResource.KnownParameters.PrincipalType)
+                      .WithManifestPublishingCallback(resource.WriteToManifest);
+    }
+
+    /// <summary>
+    /// Adds an Azure OpenAI Deployment resource to the application model. This resource requires an <see cref="AzureOpenAIResource"/> to be added to the application model.
+    /// </summary>
+    /// <param name="serverBuilder">The Azure SQL Server resource builder.</param>
+    /// <param name="name">The name of the deployment.</param>
+    /// <param name="modelName">The model name.</param>
+    /// <param name="modelVersion">The model version.</param>
+    /// <param name="skuName">The sku name.</param>
+    /// <param name="skuCapacity">Teh sku capacity.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{AzureSqlDatabaseResource}"/>.</returns>
+    public static IResourceBuilder<AzureOpenAIDeploymentResource> AddDeployment(this IResourceBuilder<AzureOpenAIResource> serverBuilder, string name, string modelName, string modelVersion, string skuName = "Standard", int skuCapacity = 2)
```

Comment:
```suggestion
    public static IResourceBuilder<AzureOpenAIDeploymentResource> AddDeployment(this IResourceBuilder<AzureOpenAIResource> builder, string name, string modelName, string modelVersion, string skuName = "Standard", int skuCapacity = 2)
```

What does `serverBuilder` mean here?